app/src/main/java/de/danoeh/antennapod/adapter/actionbutton/ItemActionButton.java

```diff
@@ -33,18 +33,22 @@ public int getVisibility() {
     }
 
     @NonNull
-    public static ItemActionButton forItem(@NonNull FeedItem item, boolean isInQueue, boolean allowStream) {
+    public static ItemActionButton forItem(@NonNull FeedItem item, boolean allowStream) {
         final FeedMedia media = item.getMedia();
         if (media == null) {
             return new MarkAsPlayedActionButton(item);
         }
 
         final boolean isDownloadingMedia = DownloadRequester.getInstance().isDownloadingFile(media);
+        final boolean isInQueue = item.isTagged(FeedItem.TAG_QUEUE);
```

Comment:
How about simply removing the action button that adds to the queue? It was never shown anyway because we always passed `true`. Now, when trying to download over mobile data and selecting to add to the queue, the list still displays download buttons but when scrolling, it replaces them with queue buttons. Even if the list was updated, I think it would be more clear to just continue displaying the download button.

Reply:
Yea also with SwipeActions coming up theres no need for a queue button.